src/app/reporting/Engine.cpp

```diff
@@ -47,6 +47,31 @@ void Engine::Shutdown()
     mGlobalDirtySet.ReleaseAll();
 }
 
+bool Engine::IsClusterDataVersionMatch(ClusterInfo * aDataVersionFilterList, const ConcreteReadAttributePath & aPath)
+{
+    bool existPathMatch       = false;
+    bool existVersionMismatch = false;
+    for (auto filter = aDataVersionFilterList; filter != nullptr; filter = filter->mpNext)
+    {
+        if (aPath.mEndpointId == filter->mEndpointId && aPath.mClusterId == filter->mClusterId)
+        {
+            existPathMatch = true;
+        }
+
+        if (aPath.mEndpointId == filter->mEndpointId && aPath.mClusterId == filter->mClusterId &&
+            !(filter->mDataVersion.HasValue() &&
+              IsClusterDataVersionEqual(filter->mEndpointId, filter->mClusterId, filter->mDataVersion.Value())))
+        {
+            existVersionMismatch = true;
+        }
+    }
+    if (!existPathMatch || existVersionMismatch)
+    {
+        return false;
+    }
+    return true;
```

Comment:
Might be simpler as:
```suggestion
    return existPathMatch && !existVersionMismatch;
```

Reply:
done, thanks

Reply:
Doesn't look done...

Reply:
yep, something wrong happens in rebase...just update it, thanks